### PR TITLE
Improves interface between processor and commands.

### PR DESCRIPTION
**Pull request by vladlosev**
@parkr This refactoring changes commands into objects and simplifies both the input API (`getCollection`) and output API (returning response objects, `ensureCollection`). Tests are not affected.

### CODE REVIEW

File: lib/commands/base.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+function BaseCommand(server) {
+  this.server = server;
+}
+
+BaseCommand.prototype.getDbAndCollectionName = function(clientReqMsg) {
+   var fullCollectionName = clientReqMsg.fullCollectionName;
+   var firstDot = fullCollectionName.indexOf('.');
```
Comment (parkr) [inline lib/commands/base.js]: I wish you had lodash :(

http://jsperf.com/lodash-contains-vs-native-indexof/3

File: lib/commands/count.js [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
-var protocol = require('../protocol');
+var BaseCommand = require('./base');
+
+function Count(server) {
+  BaseCommand.call(this, server);
+  this.logger = this.server.logger;
+}
+util.inherits(Count, BaseCommand);
+Count.commandName = 'count';
+Count.prototype.commandName = 'count';
```
Comment (parkr) [inline lib/commands/count.js]: this is awkwardly duplicated, it feels like it could be derived somehow

File: lib/commands/base.js (hunk 1 above)
Comment (vladlosev) [inline lib/commands/base.js]: I can haz lodash but this is `indexOf` on a string.

File: lib/commands/count.js (hunk 2 above)
Comment (vladlosev) [inline lib/commands/count.js]: DRYed them out a bit.

**Pull request merged**